File: Resources/luaScript/util/HXUtil.lua
--[[=============================================================================
#     FileName: HXUtil.lua
#         Desc: 工具函数
#       Author: hanxi
#        Email: hanxi.com@gmail.com
#     HomePage: http://hanxi.cnblogs.com
#      Version: 0.0.1
#   LastChange: 2013-08-22 10:08:40
#      History:
=============================================================================]]

-- for CCLuaEngine traceback
function __G__TRACKBACK__(msg)
    cclog("----------------------------------------")
    cclog("LUA ERROR: " .. tostring(msg) .. "\n")
    cclog(debug.traceback())
    cclog("----------------------------------------")
end

CCLuaLog = function(...)
    print(...)
end

--合并集合,根据key值
function HXUtilMergeSet(set1,set2)
    for k,v in pairs(set2) do
        set1[k] = v
    end
end

File: HXModules/HXUtil/HXUtil.cpp
/*=============================================================================
#     FileName: HXUtil.cpp
#         Desc: engine system
#       Author: hanxi
#        Email: hanxi.com@gmail.com
#     HomePage: http://hanxi.cnblogs.com
#      Version: 0.0.1
#   LastChange: 2013-10-02 17:00:35
#      History:
=============================================================================*/
#include "HXUtil.h"

USING_NS_CC;

namespace hx{ namespace hxutil {

long HXMillisecondNow() {
    struct timeval tv;
    gettimeofday(&tv,NULL);
    return tv.tv_sec * 1000 + tv.tv_usec / 1000;
}

RenderTexture* HXSaveScreenToRenderTexture()
{
    Size winSize = Director::getInstance()->getWinSize();
    RenderTexture* render  = RenderTexture::create(winSize.height, winSize.width);

    render->begin();
    Director::getInstance()->drawScene();
    render->end();

    return render;
}

}} // end namespace hx namespace hxutil

